### PR TITLE
Fix storage emulator for file.delete() API

### PR DESCRIPTION
**Pull request by jpreynat**
### Description

The current storage emulator doesn't support using `@google-cloud/storage`'s API to delete a file directly.
Instead, it fails with a `Not Found` error.

This PR fixes this by updating the `gcloud` emulator API `DELETE` endpoint to not decode the decode the `objectId` and fix the parameter name in the router, like for other routes.

### Scenarios Tested

```ts
import { Storage } from '@google-cloud/storage';

const storage = new Storage({
    credentials: // ...
});

const bucket = storage.bucket('my-bucket');

const file = bucket.file('path/to/my/file');

file.delete().then(() => {
    // Never reaches here as the method throws even when the file exists
}));
```

I added a test that failed before updating the `gcloud` API with the error:
```
Not Found
```

After the update, it passes properly 👌

### CODE REVIEW

File: scripts/storage-emulator-integration/tests.ts
Comment (yuchenshi) [inline scripts/storage-emulator-integration/tests.ts:256]: This also passes if file exists. To mitigate this just assert file doesn't exist (!fs.existsSync).

FYI, in general testing for errors should use the formula `
expect(func).to.throw().with.property('code', 'ENOENT');`

File: src/emulator/storage/apis/gcloud.ts
Comment (yuchenshi) [inline src/emulator/storage/apis/gcloud.ts:94]: Do we need to decode? I think the best way to find out is to add a test with a file under a nested path (e.g. `a/b/c.txt`) and delete that

Comment (abeisgoat) [reply src/emulator/storage/apis/gcloud.ts:94]: +1 I'm pretty sure decoding is needed

Comment (jpreynat) [reply src/emulator/storage/apis/gcloud.ts:94]: Decoding was one of the reason why it didn't work.
If you look at the other route handlers (get and patch for instance), they don't apply decoding.

File: scripts/storage-emulator-integration/tests.ts
Comment (jpreynat) [reply scripts/storage-emulator-integration/tests.ts:256]: Sure, I'll use this check instead 👍

Comment (jpreynat) [reply scripts/storage-emulator-integration/tests.ts:256]: I also see that I forgot to remove the `it.only` on the test.
I'll remove that at the same time.

File: src/emulator/storage/apis/gcloud.ts
Comment (abeisgoat) [reply src/emulator/storage/apis/gcloud.ts:94]: Can you add a test of deleting at a nested path so we know the behavior is verified?

Comment (jpreynat) [reply src/emulator/storage/apis/gcloud.ts:94]: I actually rewrote the whole test to make sure to use a nested path and that the file exists and is then properly deleted.
I was also foolishly testing if the file was deleted on `fs` instead of on the bucket. This is now checked properly.
Let me know what you think.

Comment (abeisgoat) [reply src/emulator/storage/apis/gcloud.ts:94]: looks great! I think we'll be good to merge unless Yuchen has any issues

Comment (jpreynat) [reply src/emulator/storage/apis/gcloud.ts:129]: @abeisgoat @yuchenshi
Just updated the branch to resolve the conflicts after my other PR was merged.

Comment (yuchenshi) [reply src/emulator/storage/apis/gcloud.ts:129]: Good to go.

**Review by yuchenshi — APPROVED** [review]

**Comment by yuchenshi** [comment]
Thanks for the PR and making the changes! This is merged and will be included in the next release.

**Comment by jpreynat** [comment]
Perfect, thanks for the quick review 👌

**Pull request merged**
